front/src/applications/editor/tools/speedSectionEdition/components.tsx

```diff
@@ -186,37 +211,179 @@ export const MetadataForm: FC = () => {
+                  );
+                  setState({ entity: newEntity });
+                }}
+              />
+              <input
+                className="form-control flex-shrink-0 px-2"
+                style={{ width: '5em' }}
+                placeholder=""
+                type="number"
+                min={0}
+                step={0.1}
+                value={msToKmh(value)}
+                onChange={(e) => {
+                  const newEntity = cloneDeep(entity);
+                  newEntity.properties.speed_limit_by_tag =
+                    newEntity.properties.speed_limit_by_tag || {};
+                  newEntity.properties.speed_limit_by_tag[key] = kmhToMs(
+                    parseFloat(e.target.value)
+                  );
+                  setState({ entity: newEntity });
+                }}
+              />
+              <span className="text-muted ml-2">km/h</span>
+              <small>
+                <button
+                  type="button"
+                  className="btn btn-primary btn-sm px-2 ml-2"
+                  title={t('commons.delete')}
+                  onClick={() => {
+                    const newEntity = cloneDeep(entity);
+                    newEntity.properties.speed_limit_by_tag = omit(
+                      newEntity.properties.speed_limit_by_tag || {},
+                      key
+                    );
+                    setState({ entity: newEntity });
+                  }}
+                >
+                  <FaTimes />
+                </button>
+              </small>
+            </div>
+          </div>
+        ))}
+        <button
+          type="button"
+          className="btn btn-secondary w-100 text-wrap small mb-2"
+          onClick={async () => {
+            const newEntity = cloneDeep(entity);
+            newEntity.properties.speed_limit_by_tag = newEntity.properties.speed_limit_by_tag || {};
+            let key = t('Editor.tools.speed-edition.new-tag');
+            let i = 1;
+            if (newEntity.properties.speed_limit_by_tag[key]) {
+              while (newEntity.properties.speed_limit_by_tag[`${key} ${i}`]) i += 1;
+              key += ` ${i}`;
+            }
+            newEntity.properties.speed_limit_by_tag[key] = kmhToMs(80);
+            setState({ entity: newEntity });
+          }}
+        >
+          <AiOutlinePlusCircle className="mr-2" />
+          {t('Editor.tools.speed-edition.add-new-speed-limit')}
+        </button>
       </form>
     </div>
   );
 };
 
 export const SpeedSectionEditionLeftPanel: FC = () => {
+  const dispatch = useDispatch();
   const { t } = useTranslation();
+  const {
+    setState,
+    state: { entity, initialEntity },
+  } = useContext(EditorContext) as ExtendedEditorContextType<SpeedSectionEditionState>;
+  const isNew = entity.properties.id === NEW_ENTITY_ID;
+  const [isLoading, setIsLoading] = useState(false);
 
   return (
     <div>
       <legend>{t('Editor.obj-types.SpeedSection')}</legend>
```

Comment:
In valid usage of legend, it can be only used as first child of a fieldset tag 
https://html.spec.whatwg.org/multipage/form-elements.html#the-legend-element